statsmodels/discrete/conditional_models.py

```diff
@@ -582,13 +582,20 @@ def loglike(self, params):
         lpr = np.dot(self.exog, pmat)
 
         ll = 0.0
+
+        # denom - immediately calculate the sums for the selected elements,
+        # obtaining exponential values, ultimately obtaining a final
+        # calculating the total amount
+
         for ii in self._grp_ix:
             x = lpr[ii, :]
             jj = np.arange(x.shape[0], dtype=int)
             y = self.endog[ii]
-            denom = 0.0
-            for p in itertools.permutations(y):
-                denom += np.exp(x[(jj, p)].sum())
+            denom = np.sum(
+                np.exp(
+                    np.sum(x[jj, list(itertools.permutations(y))], axis=1)
+                )
```

Comment:
You could remove the outer sum and exp and use scipy.special.logsumexp below for a little better stability.

```python
            denom = np.sum(x[jj, list(itertools.permutations(y))], axis=1)
            ll += x[(jj, y)].sum() - logsumexp(denom)
```

Reply:
@jseabold updated the code. I posted the code for the tests in a separate message.